### PR TITLE
move container files under /run writeable storaage

### PR DESCRIPTION
**Pull request by katiewasnothere**
related to https://github.com/microsoft/hcsshim/pull/731 

Signed-off-by: Kathryn Baldauf <kabaldau@microsoft.com>

### CODE REVIEW

File: client/process.go [hunk 1]
```diff
@@ -134,16 +134,16 @@ func (config *Config) DebugGCS() {
...
 		cmd = `sh -c "`
 		cmd += debugCommand("kill -10 `pidof gcs`") // SIGUSR1 for stackdump
-		cmd += debugCommand("ls -l /tmp")
-		cmd += debugCommand("cat /tmp/gcs.log")
-		cmd += debugCommand("cat /tmp/gcs/gcs-stacks*")
-		cmd += debugCommand("cat /tmp/gcs/paniclog*")
-		cmd += debugCommand("ls -l /tmp/gcs")
-		cmd += debugCommand("ls -l /tmp/gcs/*")
-		cmd += debugCommand("cat /tmp/gcs/*/config.json")
+		cmd += debugCommand("ls -l /run")
```
Comment (jterry75) [inline client/process.go]: Just remove these changes since client is going away anyways.

File: init/init.c [hunk 2]
```diff
@@ -31,7 +31,7 @@ const char *const default_envp[] = {
 };
 
 // When nothing is passed, default to the LCOWv1 behavior.
-const char *const default_argv[] = { "/bin/gcs", "-loglevel", "debug", "-logfile=/tmp/gcs.log" };
+const char *const default_argv[] = { "/bin/gcs", "-loglevel", "debug", "-logfile=/run/gcs.log" };
```
Comment (jterry75) [inline init/init.c]: `/run/gcs/gcs.log`

File: internal/runtime/hcsv2/standalone_container.go [hunk 3]
```diff
@@ -15,8 +15,7 @@ import (
 )
 
 func getStandaloneRootDir(id string) string {
-	return filepath.Join("/tmp/gcs/s", id)
-	return filepath.Join("/tmp/gcs/s", id)
+	return filepath.Join("/run/gcs/c", id)
```
Comment (jterry75) [inline internal/runtime/hcsv2/standalone_container.go]: What about `workload_container.go`?

File: service/gcs/main.go [hunk 4]
```diff
@@ -33,7 +33,7 @@ func main() {
 		fmt.Fprintf(os.Stderr, "\nUsage of %s:\n", os.Args[0])
 		flag.PrintDefaults()
 		fmt.Fprintf(os.Stderr, "Examples:\n")
-		fmt.Fprintf(os.Stderr, "    %s -loglevel=debug -logfile=/tmp/gcs.log\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "    %s -loglevel=debug -logfile=/run/gcs.log\n", os.Args[0])
```
Comment (jterry75) [inline service/gcs/main.go]: Same `/run/gcs/gcs.log`

File: client/process.go (hunk 1 above)
Comment (katiewasnothere) [reply client/process.go]: Done.

File: init/init.c (hunk 2 above)
Comment (katiewasnothere) [reply init/init.c]: All changes related to this are done as well.

File: service/gcs/runtime/runc/utils.go [hunk 5]
```diff
@@ -61,9 +61,13 @@ func (r *runcRuntime) makeContainerDir(id string) error {
 	return nil
 }
 
+func (r *runcRuntime) getContainerStateDir(id string) string {
+	return filepath.Join(r.runcLogBasePath, id, "state")
```
Comment (jterry75) [inline service/gcs/runtime/runc/utils.go]: So matching the host changes I think this should now be `/run/gcs/c/<id>/bundle`. IE: runc logs get written to the runc OCI bundle dir for the container. Seems like a nice place to find them.

**Pull request merged**